aptos-move/framework/src/extended_checks.rs

```diff
@@ -498,6 +498,31 @@ impl<'a> ExtendedChecker<'a> {
                 self.env
                     .error(&fun.get_id_loc(), "view function must return values")
             }
+
+            fun.get_parameter_types()
+                .iter()
+                .for_each(|parameter_type| match parameter_type {
+                    Type::Primitive(inner) => {
+                        if inner == &PrimitiveType::Signer {
+                            self.env.error(
+                                &fun.get_id_loc(),
+                                "view function cannot use the signer paremter",
+                            )
+                        }
+                    },
+                    Type::Reference(_, inner) => {
+                        if let Type::Primitive(inner) = inner.as_ref() {
+                            if inner == &PrimitiveType::Signer {
+                                self.env.error(
+                                    &fun.get_id_loc(),
+                                    "view function cannot use the & signer paremter",
+                                )
+                            }
+                        }
+                    },
+                    _ => (),
+                });
```

Comment:
Hmm... wouldn't this need to be feature gated?

Reply:
@runtian-zhou Ah I see that this function is indeed not called from the VM, so feature gating may not be necessary. However let me also check the extended checks we perform at publishing time and make sure we're doing the right thing there -- if we are disallow view functions with signers then we should also block it at the VM level

Reply:
Yea I think we are fine here. The extended check uses move-model which we won't have it during publishing time for sure.

Reply:
The on chain check is performed here: https://github.com/aptos-labs/aptos-core/blob/main/aptos-move/framework/src/module_metadata.rs#L366

Reply:
cc @zekun000 see discussion here.